### ISSUE TITLE
Make Layout/EmptyLineAfterGuardClause accept `:nocov:` directive after guard clause

### ISSUE THREAD

**Issue by G-Rath** [rubocop/rubocop#12151]
## Is your feature request related to a problem? Please describe.

I expect the cop not the flag the case where the line after the guard clause is a `:nocov:` directive e.g.

```ruby
allow(Open3).to receive(:capture3).and_wrap_original do |original_method, *args|
  # :nocov:
  raise "unexpected Open3.capture3 call" unless some_condition
  # :nocov:

  original_method.call(*args)
end
```

## Describe the solution you'd like

I'd like RuboCop to allow the above

## Describe alternatives you've considered

Living with this:

```
allow(Open3).to receive(:capture3).and_wrap_original do |original_method, *args|
  # :nocov:
  raise "unexpected Open3.capture3 call" unless some_condition

  # :nocov:

  original_method.call(*args)
end
```

## Additional context

This is effectively the same as https://github.com/rubocop/rubocop/issues/9146 - I'm happy to do a PR implementing this.
